lib/plsql/connection.rb

```diff
@@ -197,9 +197,8 @@ def session_id
       @session_id ||= select_first("SELECT TO_NUMBER(USERENV('SESSIONID')) FROM dual")[0]
     end
 
-    # Set time zone (default taken from TZ environment variable)
+    # Set time zone
     def set_time_zone(time_zone=nil)
-      time_zone ||= ENV['TZ']
```

Comment:
ORA_SDTZ may hold some values (OS_TZ, DB_TZ) which are not eligible for time_zone db parameter.

Let's think about possible options:
* Allow `nil` but do nothing (like your initial version)
* Or raise an exception (e.g. by passing it through and let Oracle raise error)
* Overwrite to mean 'local'
* Overwrite with ENV['TZ'] environment var - as it used to be. (Put a remark that this is not encouraged since tz names may be different in Oracle).
* Overwrite with ENV['RUBY_PLSQL_TZ']